app/src/main/java/org/fossasia/phimpme/gallery/activities/LFMainActivity.java

```diff
@@ -3496,6 +3512,95 @@ public void run() {
+            NotificationHandler.make(R.string.Images, R.string.zip_fol, R.drawable.ic_archive_black_24dp);
+        }
+
+        @Override
+        protected String doInBackground(Void... voids) {
+            DateFormat dateFormat = new SimpleDateFormat("ddMMyy_HHmm");
+            String date = dateFormat.format(Calendar.getInstance().getTime());
+
+            try {
+                double c = 0.0;
+                File file = new File(Environment.getExternalStorageDirectory() + "/" + "Phimpme_ImageZip");
+                FileOutputStream dest = null;
+                if(file.exists() && file.isDirectory()){
+                    try{
+                        dest = new FileOutputStream(file.getPath() + "/" + "ZIP_"+date+".zip");
+
+                    }catch(Exception e){
+                        e.printStackTrace();
+                    }
+                }else {
+                    if (file.mkdir()) {
+                         dest = null;
+                        try {
+                            dest = new FileOutputStream(file.getPath() + "/" + "ZIP_"+date+".zip");
+                        } catch (Exception e) {
+                            e.printStackTrace();
+                        }
+                    }
+                }
+
+                BufferedInputStream origin = null;
+                ZipOutputStream out = new ZipOutputStream(new BufferedOutputStream(
+                        dest));
+                byte data[] = new byte[BUFFER];
+
+                for (int i = 0; i < path.size(); i++) {
+                    FileInputStream fi = new FileInputStream(path.get(i));
+                    origin = new BufferedInputStream(fi, BUFFER);
+
+                    ZipEntry entry = new ZipEntry(path.get(i).substring(path.get(i).lastIndexOf("/") + 1));
+                    out.putNextEntry(entry);
+                    c++;
+                    if ((int) ((c / size) * 100) > 100) {
+                        NotificationHandler.actionProgress((int) c, path.size(), 100, R.string.zip_operation);
+                    } else {
+                        NotificationHandler.actionProgress((int) c, path.size(), (int) ((c / path.size()) * 100), R.string
+                                .zip_operation);
+                    }
+                    int count;
+
+                    while ((count = origin.read(data, 0, BUFFER)) != -1) {
+                        out.write(data, 0, count);
+                    }
+                    origin.close();
+                }
+                out.close();
+                if (isCancelled()) {
+                    return null;
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+            return date;
+        }
+
+        @Override
+        protected void onPostExecute(String date) {
+            super.onPostExecute(date);
+            NotificationHandler.actionPassed(R.string.zip_completion);
+            String path = "ZIP: "+date+".zip";
```

Comment:
This meaning I cant create two zips on the same day?

Reply:
@abishekvashok  oh i'm sorry i had used the date for testing purpose but didn't remember before pushing the code. I will add time too